sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/AttributeSet.scala

```diff
@@ -104,13 +104,19 @@ class AttributeSet private (private val baseSet: mutable.LinkedHashSet[Attribute
    * in `other`.
    */
   def --(other: Iterable[NamedExpression]): AttributeSet = {
```

Comment:
@peter-toth  What about changing the `other` to a call-by-name parameter?
```suggestion
  def --(other: => Iterable[NamedExpression]): AttributeSet = {
```

Reply:
and then we can save here what in the `missingInput()` was saved in your previous commit (the calculation of the `inputSet`)

Reply:
This can be more efficient, but looks weird in standard collection APIs such as `def --`